disk-utils/mkfs.bfs.c

```diff
@@ -260,13 +264,29 @@ int main(int argc, char **argv)
 	sb.s_start = cpu_to_le32(ino_bytes + sizeof(struct bfssb));
 	sb.s_end = cpu_to_le32(total_blocks * BFS_BLOCKSIZE - 1);
 	sb.s_from = sb.s_to = sb.s_backup_from = sb.s_backup_to = -1;
-	memcpy(sb.s_fsname, fsname, 6);
-	memcpy(sb.s_volume, volume, 6);
+
+	if (fsname == NULL){
+		fsname = "      ";
+		memcpy(sb.s_fsname, fsname, 6);
+	}
+	else{
+		memcpy(sb.s_fsname, fsname, 6);
+		free(fsname);
+	}
+	if (volume == NULL){
+		volume = "      ";
+		memcpy(sb.s_volume, volume, 6);
+	}
+	else{
+		memcpy(sb.s_volume, volume, 6);
+		free(volume);
+	}
+	volume = fsname = NULL;
 
 	if (verbose) {
 		fprintf(stderr, _("Device: %s\n"), device);
-		fprintf(stderr, _("Volume: <%-6s>\n"), volume);
-		fprintf(stderr, _("FSname: <%-6s>\n"), fsname);
+		fprintf(stderr, _("Volume: <%-6s>\n"), sb.s_volume);
+		fprintf(stderr, _("FSname: <%-6s>\n"), sb.s_fsname);
```

Comment:
Here, we cannot be sure it's zero-terminated if all 6 bytes of the names are filled (e.g. 'foobar'). You have to use ".6%".